src/maybe_dangling.rs

```diff
@@ -25,11 +25,133 @@ use ::core::mem::ManuallyDrop as StdMD;
 ///
 /// For the adventurous `nightly` users, you can enable the
 /// `nightly-dropck_eyepatch` Cargo feature to opt into the usage of the
-/// eponymous `rustc` feature so as to get the `Drop` impl amended accordingly.
+/// [eponymous `rustc` feature][RFC-1327] so as to get the `Drop` impl amended
+/// accordingly.
+///
+/// Below are three scenarios that should illustrate how Drop Check (`dropck`)
+/// handles transitive drop glue when checking for dangling pointers.
+/// All three scenarios are theoretically sound, since no dangling pointer is
+/// ever accessed in any destructor.
+/// But none will compile without the `nightly-dropck_eyepatch` feature there
+/// to relax Drop Check into thinking that we won't accidentially access a
+/// dangling pointer.
+/// The `nightly-dropck_eyepatch` provides the flexibility to make the first
+/// two cases compile.
+///
+/// #### Scenario 1: `T` has no destructor
+///
+/// With `#[may_dangle]` we are able to communicate to Drop Check that
+/// `MaybeDangling` won't access the potentially dangling `T` in its destructor,
+/// *unless* `T` is invovled in transitive drop glue, i.e. `T` implements `Drop`
+/// itself.
+/// Since `T` does not implement `Drop`, Drop Check will allow this to compile,
+/// even though the reference stored in `Wrapper` is dangling when `v` gets
+/// dropped:
+///
+/// ```
+/// # #[cfg(feature = "nightly-dropck_eyepatch")]
+/// # {
+/// use maybe_dangling::MaybeDangling;
```

Comment:
```suggestion
/// use ::maybe_dangling::MaybeDangling;
```

This is a very personal nit but which I care about for crates I work with: I like to make it obvious in examples and snippets whether we are talking of some external crate (such as ourselves), or some module potentially in scope and which thus requires the reader to keep looking in order to sort it out